src/datachain/lib/file.py

```diff
@@ -283,9 +282,8 @@ def get_uri(self):
     def get_path(self) -> str:
         """Returns file path."""
         path = unquote(self.get_uri())
-        fs = self.get_fs()
-        if isinstance(fs, LocalFileSystem):
-            # Drop file:// protocol
+        source = urlparse(self.source)
+        if source.scheme == "file":
```

Comment:
@ilongin, is there any reason why we keep drive in `source`? I think all of this would be much simpler if we had everything we need in `parent + name` (well, soon that will be called `path`).

Reply:
Also, if we want to support s3-compatible storage and use of multiple storages at the same time, we need to also preserve `fsid` or endpointurl. Otherwise, how will we know if the path is from AWS s3 or some compatible storage?

Reply:
Having `file:///` as source was part of the solution to remove old  `add_storage` method that was needed to be called before being able to index local FS, and to have things like partial indexing for local FS indexes (avoid re-indexing already indexed folders). The idea was to look local FS similar as one bucket and every directory in it is similar to directories in cloud buckets. This way every time someone tries to index some dir, it will create another partial index and can check if that dir is already indexed in some other partial index. Also, with this user will have only one entry in `buckets` table for local FS and that is the root of the system itself (`file:///`).

Reply:
Sorry, I should have been clearer. We have `file:///<drive>/` (`file:///C:/`) as source for Windows.